src/components/singleReview.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getReviewById, addVotes } from "../api";
+
+const SingleReview = () => {
+    const { review_id } = useParams()
+    const [review, setReview] = useState([]);
+    const [userVote, setUserVote] = useState(0);
+
+    const handleUpvote = () => {
+        setUserVote((prevVote) => prevVote + 1);
+        addVotes(1, review_id)
+    };
+
+    const handleDownvote = () => {
+        setUserVote((prevVote) => prevVote - 1);
+        addVotes(-1, review_id)
```

Comment:
Id like to see a way to limit our users to only be able to add or subtract one vote on an article instead of being able to add as many votes as they like. 

Id also like to see some error handling. What should the user see if the API call to patch the vote doesn't go through properly?